Makefile.include

```diff
@@ -10,6 +10,13 @@ RIOTCPU := $(abspath $(RIOTCPU))
 RIOTBOARD ?= $(RIOTBASE)/boards
 RIOTBOARD := $(abspath $(RIOTBOARD))
 
+# Top level git root of the project being built
+RIOTPROJECT ?= $(shell git rev-parse --show-toplevel)
+RIOTPROJECT := $(abspath $(RIOTPROJECT))
```

Comment:
When building external projects RIOTBASE will not be the git root.
an example would be building stuff in https://github.com/RIOT-OS/applications with the command `make RIOTBASE=/home/asdf/RIOT`